src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -69,29 +69,164 @@ on_exit() {
     set -xe
   fi
 
+  # backup cache
+  if [[ $preserve_cache == true ]]; then
+    echo "=== Execution complete — caching current session to $DX_CACHEDIR/$NXF_UUID"
+
+    # wrap cache folder and upload cache.tar
+    if [[ -n "$(ls -A .nextflow)" ]]; then
+      tar -cf cache.tar .nextflow
+
+      CACHE_ID=$(dx upload "cache.tar" --path "$DX_CACHEDIR/$NXF_UUID/cache.tar" --no-progress --brief --wait -p -r) &&
+        echo "Upload cache of current session as file: $CACHE_ID" &&
+        rm -f cache.tar ||
+        echo "Failed to upload cache of current session $NXF_UUID"
+    else
+      echo "No cache is generated from this execution. Skip uploading cache."
+    fi
 
-  # remove .nextflow from the current folder /home/dnanexus/output_files
+  # preserve_cache is false
+  # clean up files of this session
+  else
+    echo "=== Execution complete — cache and working files will not be resumable"
+  fi
+
+  # remove .nextflow from the current folder /home/dnanexus/nextflow_execution
   rm -rf .nextflow
+  rm nxf_runtime.config
 
   # try uploading the log file if it is not empty
   if [[ -s $LOG_NAME ]]; then
-    mkdir ../nextflow_log
-    mv $LOG_NAME ../nextflow_log/$LOG_NAME || true
+    mkdir -p /home/dnanexus/out/nextflow_log
+    mv "$LOG_NAME" "/home/dnanexus/out/nextflow_log/$LOG_NAME" || true
```

Comment:
do we need "|| true"? If this fails, do we want to inform the user instead?

Reply:
can this mv even fail? As we check whether the log file exists/is not empty?

Reply:
Since the nextflow execution is completed, I think it is better to make nothing fail (otherwise the user might wait for hours but end up fail the head job with a minor error.) But you are right I dont think it would/should fail.